jobs/jobs.go

```diff
@@ -286,16 +288,18 @@ func (job *Job) processEvent(ctx context.Context, event events.Event) bool {
 			break
 		}
 		log.Debugf("job exited but restart not permitted: %v", job.Name)
+		job.startEvent = events.NonEvent
 		return true
 	case job.startEvent:
 		if job.startsRemain == 0 {
+			job.startEvent = events.NonEvent
```

Comment:
Is this if clause necessary? Since if we hit `startsRemain == 0` below, the event will be overwridden which will prevent further entries into the `case job.startEvent`.